### PR TITLE
Modified Columns in APIs and made changes to the mutations accordingly

### PR DESCRIPTION
**Pull request by ganatra-v**
## Pull request checklist

Please check if your PR :rocket: fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run start`) was run locally and new changes were working correctly
- [x] Lint (`npm run linter`) has passed locally and any fixes were made for failures

## Pull request type

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Fixes #

## What is the new behavior?

- Added Columns to the entities as per the requirement
- Modified the queries and mutations as per the new columns

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

The name of the mutations and queries have been modified.

### CODE REVIEW

File: src/database/entity/Card.ts [hunk 1]
```diff
@@ -39,27 +38,42 @@ export class Card {
...
+	@ManyToOne((type) => Users, user => user.created_cards)
+	created_by!: Users;
 
 	@Column('text', { array: true, nullable: true })
 	@Field(() => [String], {
 		nullable: true
 	})
 	labels?: string[];
 
+	@Column('text', { array: true, nullable: true })
```
Comment (Sarvesh67) [inline src/database/entity/Card.ts]: Why is label color an individual field? Labels should be stored as an array of {description, color}

File: src/database/entity/Card.ts [hunk 2]
```diff
@@ -39,27 +38,42 @@ export class Card {
...
+	@Field(() => [String], {
+		nullable:true
+	})
+	tasks?: string[];
+
+	@Column('boolean', { array: true, nullable: true })
+	@Field(() => [Boolean], {
+		nullable:true
+	})
+	task_status?: boolean[];
```
Comment (Sarvesh67) [inline src/database/entity/Card.ts]: Similarly, why is task_status an individual field? Tasks (or checklist - its a better name as it retains ambiguity) should be an array of {description, status}.

File: src/database/entity/Card.ts [hunk 3]
```diff
@@ -70,17 +84,20 @@ export class Card {
...
 	members?: Users[];
 
 	@Column({
 		type: 'timestamp',
 		nullable: true
 	})
-	// @Field()
-	completed_at?: Date;
+	@Field()
+	completed_at?: string;
```
Comment (Sarvesh67) [inline src/database/entity/Card.ts:130]: Reason for change in type?

File: src/resolvers/ProjectResolver.ts [hunk 4]
```diff
@@ -27,33 +27,38 @@ class ProjectInput {
...
 }
 
 @Resolver(Project)
 export class ProjectResolver {
 	ProjectRepo = getCustomRepository(ProjectRepository);
 
 	@Mutation(() => Project)
-	async createProject(
-		@Arg('project_data') { name, description, created_by_id }: ProjectInput
+	async CreateProject(
```
Comment (Sarvesh67) [inline src/resolvers/ProjectResolver.ts:40]: In here, pls create four new task boards (default task boards) - ongoing, todo, completed, bugs/fixes

**Review by Sarvesh67 — COMMENTED** [review]
If changes made to entities, make appropriate changes to resolvers and repositories

File: src/database/entity/Card.ts (hunk 3 above)
Comment (ganatra-v) [reply src/database/entity/Card.ts:130]: On running it with Date type, GraphQL gave an error message, 
`Unable to serialize value '2021-01-23T04:05:11.922Z' as it's not an instance of 'Date'` 
on retrieving the `completed_at` field for completed cards. So, I changed the type to `string` and now it is working smoothly.

File: src/resolvers/ProjectResolver.ts (hunk 4 above)
Comment (ganatra-v) [reply src/resolvers/ProjectResolver.ts:40]: The idea in mind was, every project, say Mello, will have multiple taskboards, say 1 board for the FrontEnd, another for the BackEnd. 
Each Card has a `card_status` of `ongoing`, `todo` or `completed`, which should be displayed in the required manner by the front-end. If required, I can add `bug` in the enum for `card_status`

File: src/database/entity/Card.ts (hunk 1 above)
Comment (ganatra-v) [reply src/database/entity/Card.ts]: I have modified the storage of labels. The description and color are now being stored as a single object in the database

**Pull request merged**